### PR TITLE
Prevent invalid url from raising DataStructureException

### PR DESCRIPTION
**Pull request by amdomanska**
Fixes: DOAJ/doajPM#2038
Continuation: https://github.com/DOAJ/doaj/pull/1660

Changes made according to this comment: https://github.com/DOAJ/doaj/pull/1660#pullrequestreview-274292043

### CODE REVIEW

**Comment by amdomanska** [comment]
Sorry for the new PR. The previous one got closed automatically by GH when I reverted branch's state to the original.

**Comment by amdomanska** [comment]
@richard-jones I've tried that for a few documents (both with correct and incorrect URLs) and confirmed with the existing set of UT.

**Comment by richard-jones** [comment]
@Steven-Eardley - I think we can queue this one up for deployment

File: portality/api/v1/data_objects/article.py [hunk 1]
```diff
@@ -48,14 +48,6 @@
                     }
                 },
 
-                "link": {
```
Comment (Steven-Eardley) [inline portality/api/v1/data_objects/article.py]: I think this should still be in, like:

```
"link": {
    "fields": {
        "type": {"coerce": "link_type"},
        "url": {"coerce": "unicode"},
        "content_type": {"coerce": "link_content_type"}
    }
},
```
Note `url` has been replaced by `unicode`. This is a more minimal change to the struct to allow bad URLs out of the system. I don't want to go too far down the route of making this model permissive.

In general, if we have a big problem with bad data causing 500 errors on outgoing articles, we should catch that exception when we are enforcing the outgoingarticleDO and give a JSON response saying we can't give it out.

Comment (richard-jones) [reply portality/api/v1/data_objects/article.py]: Agreed.  I pondered this same point for a while and went the other way, but if @Steven-Eardley thinks it should be in I'd +1 that.

File: doajtest/unit/test_api_crud_article.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def test_04_coerce(self):
 
         # now test some failures
 
-        # an invalid urls
+        # an invalid urls - WE DON'T WANT THIS EXCEPTION TO BE RAISED ANYMORE
```
[the remark this answers is not in the document]
Comment (amdomanska) [reply doajtest/unit/test_api_crud_article.py]: Yep, sorry, it's an oversight from the previous version that should be deleted earlier

**Comment by amdomanska** [comment]
Not a problem ofc! :)

**Comment by amdomanska** [comment]
This is the API response for retrieving the article that contains invalid URL. 

image

File: portality/api/v1/crud/articles.py [hunk 3]
```diff
@@ -123,6 +123,7 @@ def retrieve_swag(cls):
         template['responses']['200']['schema'] = IncomingArticleDO().struct_to_swag(schema_title='Article schema')
         template['responses']['401'] = cls.R401
         template['responses']['404'] = cls.R404
+        #template['responses']['500'] = cls.R500
```
Comment (Steven-Eardley) [inline portality/api/v1/crud/articles.py]: Is this supposed to be commented out?

**Pull request merged**